src/ServiceControl/Monitoring/InMemoryMonitoringDataStore.cs

```diff
@@ -71,6 +71,28 @@ public Task Delete(Guid endpointId)
             return Task.CompletedTask;
         }
 
+        public Task BulkCreate(EndpointDetails[] newEndpoints)
+        {
+            foreach (var endpoint in newEndpoints)
+            {
+                var id = DeterministicGuid.MakeId(endpoint.Name, endpoint.HostId.ToString());
+
+                if (!endpoints.Any(a => a.Id == id))
```

Comment:
Are we still assuming this stuff is never executed concurrently because we only have one batch going on and never concurrently interleaved batches? So a list is safe?

If we always end up doing `Any` on the list, would it be simpler to have a dictionary? That would also be the less allocation heavy variant because the Any will always closure allocate due to the access to the local `id`